contractrequester/contractrequester.go

```diff
@@ -238,18 +257,22 @@ func (cr *ContractRequester) ReceiveResult(ctx context.Context, parcel core.Parc
 		return nil, errors.New("ReceiveResult() accepts only message.ReturnResults")
 	}
 
-	cr.ResultMutex.Lock()
-	defer cr.ResultMutex.Unlock()
+	utils.MeasureExecutionTime(ctx, "ContractRequester.ReceiveResult",
+		func() {
+			cr.ResultMutex.Lock()
+			defer cr.ResultMutex.Unlock()
```

Comment:
it was planed, to move lock and unlock here? 
maybe need to measure lock time separately with other logic?

Reply:
I measured the time and it's neglectable. The measurement was removed afterwards because additional disk IO affects the overall performance.

Doing Lock() / Unlock() inside the lambda is exactly what I wanted. Otherwise the lock will be locked during the execution of the body of MeasureExecutionTime.